### PR TITLE
New design has been applied.

### PR DESCRIPTION
**Pull request by aleksejlopasov**
Material UI libraries now used for UI user form.
CSS and JS files moved from CDN to server static folder.
Some typos have been fixed.
HTML labels have been standardized.
Autoscaling now supports 'None' value to disable it, even if user's policies exist.

### CODE REVIEW

File: dataprocspawner/customize_cluster.py [hunk 1]
```diff
@@ -57,240 +145,787 @@ def get_custom_cluster_html_form(autoscaling_policies, node_types):
...
 
   html_custom_labels = """
-    <br />
-    <div class="form-group">
-      <label for="custom_labels">User defined labels</label>
-      <input name="custom_labels" class="form-control" placeholder="key1:value1,key2:value2"
-        value=''>
-      </input>
-    </div>"""
+    <label class="mdc-text-field jupyter-text-field mdc-text-field--outlined">
```
Comment (JerryLeiDing) [inline dataprocspawner/customize_cluster.py:787]: input label missing here https://screenshot.googleplex.com/3sbBrNmyqFJN99j

File: dataprocspawner/customize_cluster.py [hunk 2]
```diff
@@ -57,240 +145,787 @@ def get_custom_cluster_html_form(autoscaling_policies, node_types):
...
+    <br />"""
 
   html_init_actions = """
-    <div class="form-group">
-      <label for="init_actions">Initialization actions</label>
-      <input name="init_actions" class="form-control"
-        placeholder="gs://<init_action1>,gs://<init_action2>" value="">
-      </input>
-    </div>"""
+    <label class="mdc-text-field jupyter-text-field mdc-text-field--outlined">
```
Comment (JerryLeiDing) [inline dataprocspawner/customize_cluster.py:811]: Input label missing here https://screenshot.googleplex.com/3sbBrNmyqFJN99j

File: dataprocspawner/customize_cluster.py [hunk 3]
```diff
@@ -57,240 +145,787 @@ def get_custom_cluster_html_form(autoscaling_policies, node_types):
...
-    </div>
-    <div class="form-group">
-      <label for="hive_user">Hive Metastore user name</label>
-      <input name="hive_user" class="form-control" placeholder='' value=''></input>
-    </div>
-    <div class="form-group">
-      <label for="hive_passwd">Hive Metastore password</label>
-      <input name="hive_passwd" class="form-control" placeholder='' value=''></input>
-    </div>"""
+    <label class="mdc-text-field jupyter-text-field mdc-text-field--outlined">
```
Comment (JerryLeiDing) [inline dataprocspawner/customize_cluster.py:835]: All of the input labels are missing for the hive related inputs.

Also, would it be possible to group these fields under a "Hive Metastore" group like we do for the node configs?

**Review by JerryLeiDing — COMMENTED** [review]
This looks awesome! Thanks for putting this together!

I left a couple of minor comments about fields missing their labels.

On a more general note, there are two improvement I'd like to see if feasible:
1. Can we use a templating engine to generate the HTML? Each element is quite complicated but also quite repetitive, and for correctness/maintainability we should move away for manually constructing HTML.
2. How difficult would it be to have each group of inputs be hidden under it's own heading with a click-to-reveal? (Similar to how e.g. the "New Instance" page for GCE has "CPU Platform and GPU" hidden behind a dropdown. This isn't necessary for this PR, but I'd like to gauge if we can add this later.

**Comment by aleksejlopasov** [comment]
> This looks awesome! Thanks for putting this together!
> 
> I left a couple of minor comments about fields missing their labels.
> 
> On a more general note, there are two improvement I'd like to see if feasible:
> 
> 1. Can we use a templating engine to generate the HTML? Each element is quite complicated but also quite repetitive, and for correctness/maintainability we should move away for manually constructing HTML.
> 2. How difficult would it be to have each group of inputs be hidden under it's own heading with a click-to-reveal? (Similar to how e.g. the "New Instance" page for GCE has "CPU Platform and GPU" hidden behind a dropdown. This isn't necessary for this PR, but I'd like to gauge if we can add this later.

1. I agree, we need a template for each of the resources (fields, selectors, radios) and this is going to be implemented in future releases. This time we needed a way to spin up the new design very quickly.
2. I think it's possible, but we decided not to do that due to the UI is not overloaded with elements. But if you think that would be nice to hide the groups under the JS toggler, we can do that in future

File: dataprocspawner/customize_cluster.py [hunk 4]
```diff
@@ -57,240 +145,787 @@ def get_custom_cluster_html_form(autoscaling_policies, node_types):
...
+        </div>
+        <br />
+        <div class="mdc-form-field mdc-form-field-full mdc-form-field-radio">
+          <div class="mdc-radio">
+            <input
+              class="mdc-radio__native-control"
+              type="radio"
+              id="image-radio4"
+              name="image_version"
+              value="1.3-debian10"/>
```
[the remark this answers is not in the document]
Comment (aleksejlopasov) [reply dataprocspawner/customize_cluster.py]: Fixed in the latest commit.

File: dataprocspawner/customize_cluster.py (hunk 1 above)
Comment (aleksejlopasov) [reply dataprocspawner/customize_cluster.py:770]: It's not an issue, that's how outlined material text field works.
You can see demo [here](https://material-components.github.io/material-components-web-catalog/#/component/text-field)

File: dataprocspawner/customize_cluster.py (hunk 3 above)
Comment (aleksejlopasov) [reply dataprocspawner/customize_cluster.py:820]: See the comment about the labels above.
Hive settings are now grouped.

**Comment by m-mayran** [comment]
It looks really cool!

A couple of questions. 

When I run the code, I see the following behavior where:
- a subset of option is not part of CUSTOMIZE YOUR CLUSTER

Screenshot 2020-12-03 at 15 56 30

- When I click on CUSTOMISE YOUR CLUSTER, I see the `Advanced` text appearing but within the box that just toggled.

Screenshot 2020-12-03 at 15 59 37

Wondering if I am doing something wrong or if it's intended behavior.

**Comment by aleksejlopasov** [comment]
> It looks really cool!
> 
> A couple of questions.
> 
> When I run the code, I see the following behavior where:
> 
> * a subset of option is not part of CUSTOMIZE YOUR CLUSTER
> 
> * When I click on CUSTOMISE YOUR CLUSTER, I see the `Advanced` text appearing but within the box that just toggled.
> 
> Wondering if I am doing something wrong or if it's intended behavior.

I've sent you a working version, please check if you see the same there. If you reproduce the same issue every time, then I need more details from browser console. That seems like broken JS

**Comment by m-mayran** [comment]
> I've sent you a working version, please check if you see the same there. If you reproduce the same issue every time, then I need more details from browser console. That seems like broken JS

Still seeing the same. Chrome Version 87.0.4280.67 (Official Build) (x86_64)

**Pull request merged**